lib/CSSStyleDeclaration.js

```diff
@@ -1,27 +1,52 @@
 "use strict";
-var parserlib = require('./cssparser');
 
 module.exports = CSSStyleDeclaration;
 
 function CSSStyleDeclaration(elt) {
   this._element = elt;
 }
 
+const IMPORTANT_BANG = '!important';
+
 // Utility function for parsing style declarations
 // Pass in a string like "margin-left: 5px; border-style: solid"
 // and this function returns an object like
 // {"margin-left":"5px", "border-style":"solid"}
 function parseStyles(s) {
-  var parser = new parserlib.css.Parser();
-  var result = { property: Object.create(null), priority: Object.create(null) };
-  parser.addListener("property", function(e) {
-    if (e.invalid) return; // Skip errors
-    result.property[e.property.text] = e.value.text;
-    if (e.important) result.priority[e.property.text] = 'important';
-  });
-  s = (''+s).replace(/^;/, '');
-  parser.parseStyleAttribute(s);
-  return result;
+  const property = {};
```

Comment:
FYI, I think we can reuse the code of the styling parser from Angular: https://github.com/angular/angular/blob/main/packages/compiler/src/render3/view/style_parser.ts.

I believe the current implementation may miss cases where value has quotes, etc.

See more tricky cases in https://github.com/angular/angular/blob/main/packages/compiler/test/render3/style_parser_spec.ts.

Reply:
Will take a look tomorrow a